src/background/onStateChange.ts

```diff
@@ -51,6 +51,8 @@ export function onStoredStateChange(storedState: State) {
   backgroundState.showNonErrorNotifications =
     storedState.settings.notifications.enableFeedbackNotifications;
 
+  backgroundState.torrentTrackers = updateAndGetTorrentTrackers(storedState);
```

Comment:
I would prefer a shift of responsibility around this call that looks kind of like the notifications block, above. Something like:

- The background task state keeps a copy of the settings, rather than the file in `actions/`.
- When the settings changes, it's this code (and not `updateAndGetTorrentTrackers`) that decides if anything needs to be done.
- It passes the bare minimum of information to other methods that they need to do their jobs.

Reply:
So your are suggesting not the module itself but backgroundStates to hold cachedTrackersList? I'll try it.

Reply:
Yeah. `backgroundState` is the place to store anything long-lived that doesn't belong as part of the configuration of the extension. Earlier versions of the code actually used the pattern you had here, but I changed it because it got too hard to manage.